Configure

```diff
@@ -875,6 +891,17 @@ while (@argvcopy)
 		}
 	}
 
+if (grep { scalar @$_ > 0 } values %useradd) {
+    my $detected_env = join(', ', grep { @{$useradd{$_}} } sort keys %useradd);
+    if ($detected_env) {
+        die <<"_____";
+***** Mixing env / make variables and configure line compiling or linking flags
+***** is not permitted.
+***** Affected env / make variables: $detected_env
```

Comment:
Well, I'm probably not exactly in position to judge, but I find the message a bit too cryptic to unravel. Maybe "additional compiler/linker flags as command line option" would be better than "configure line compiling or linking flags"?

On side note I tried env CFLAGS=-foo-bar ./config -lfoo, and it said that affected variable is LDLIBS. It's per se correct, but maybe it would also be useful to print offending variable, one that triggered the condition.

Reply:
Good point about the message, will fix.

Regarding "offending", you do realise that CFLAGS is hardly offending for a flag that ends up in LDLIBS, right? ;-) (there's a reason I chose "affected") 
The only way to fix that is to display the name of all involved env vars that are set as well. That's a simple fix, will do.

Reply:
What I'm trying to say is if I imagine myself being a user and face the error message in question, in first hand I'd appreciate to see what is it in my environment that triggered the failure. Given the message I'd start looking for environment variable called LDLIBS, and if I don't have one, I'd be inclined to conclude that error message is not helpful.

Reply:
> The only way to fix that is to display the name of all involved env vars that are set as well.

Yes, that would do.

Reply:
> What I'm trying to say is if I imagine myself being a user and face the error message in question,

Yup, I got what you meant

Reply:
Fixed